.github/workflows/pre-commit.yml

```diff
@@ -20,27 +24,38 @@ jobs:
         uses: fkirc/skip-duplicate-actions@master
         with:
           github_token: ${{ github.token }}
-          paths: '["**.vue", "**.js", "pnpm-lock.yaml", ".github/workflows/frontendlint.yml"]'
-  test:
-    name: Frontend linting
+          paths_ignore: '["**.po", "**.json"]'
+  linting:
+    name: All file linting
     needs: pre_job
     if: ${{ needs.pre_job.outputs.should_skip != 'true' }}
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
+    - uses: actions/setup-python@v5
+      with:
+        python-version: '3.10'
     - name: Use pnpm
       uses: pnpm/action-setup@v4
     - name: Use Node.js
       uses: actions/setup-node@v4
       with:
         node-version: '18.x'
-        cache: 'pnpm'
+    - name: Get pnpm store path
+      id: pnpm-store-path
+      run: echo "dir=$(pnpm store path)" >> $GITHUB_OUTPUT
+    - name: Cache Node.js modules
+      uses: actions/cache@v4
+      with:
+        path: ${{ steps.pnpm-store-path.outputs.dir }}
+        key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
+        restore-keys: |
+          ${{ runner.os }}-pnpm-
```

Comment:
I believe the existing `cache: 'pnpm'` is a newer more suitable approach for caching. What has been added here is the 'old' way as far as I know.

https://learningequality.slack.com/archives/C0LE9NLCE/p1744912125010759

Reply:
Yeah - sorry, @akolson I should have flagged this when I recommended copy pasting from the Kolibri pre-commit action.